src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/StorePal.cs

```diff
@@ -109,7 +109,18 @@ public static IStorePal FromSystemStore(string storeName, StoreLocation storeLoc
                     if (ordinalIgnoreCase.Equals("Disallowed", storeName))
                         return AppleTrustStore.OpenStore(StoreName.Disallowed, storeLocation, openFlags);
 
-                    break;
+                    if (!IsValidStoreName(storeName))
+                        break;
+
+                    var storePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "Library", storeName + ".keychain");
+                    if (File.Exists(storePath))
+                        return AppleKeychainStore.OpenAndUnlockKeychain(storePath, openFlags);
+
+                    if ((openFlags & OpenFlags.OpenExistingOnly) == OpenFlags.OpenExistingOnly)
+                        throw new CryptographicException(SR.Cryptography_X509_StoreNotFound);
+
+                    return AppleKeychainStore.CreateKeychain(storePath, openFlags);
```

Comment:
This is going to need tests.   Since it will modify the keychain display list they should be `[OuterLoop]`; and the tests will have to P/Invoke to SecKeychainDelete since we don't have public API for that.

X509FilesystemTests.Unix.cs is the spiritual equivalent for Linux.